### PR TITLE
Setup basic backend

### PR DESCRIPTION
**Pull request by Lahphim**
Resolve https://github.com/Lahphim/go-crawler-challenge/issues/6

## What happened 👀

Setting up `postgres` with `sample` model.

## Insight 📝

- [x] postgres
- [x] basic model with migration file

Extra:

- Add a `heroku-postgresql` service to Heroku both `staging` and `production`
- Update the Dockerfile to work with migration flow by adding a migration stage and install `beego-cli` to use migrate command
- Update deploy workflow to build new `.env` file from the Github secret variables.

## Proof Of Work 📹

Add some database url to the Github secret variables.
Actions_secrets

Try to deploy and migrate some database to the staging.
Screen Shot 2564-01-19 at 15 26 11

We have 2 tables in database after migration.
pgAdmin_4-2

### CODE REVIEW

**Review by olivierobert — APPROVED** [review]

File: models/sample.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package models
+
+import (
+	"github.com/beego/beego/v2/client/orm"
+)
+
+// Sample : Sample model
+type Sample struct {
+	Id          int64  `orm:"auto"`
```
Comment (junan) [inline models/sample.go]: >  `orm:"auto"`

Do we need it? I think Beego should auto increment the number by default

File: conf/initializers/database.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	if err != nil {
+		log.Critical(fmt.Sprintf("Postgres Driver registration failed: %v", err))
+	}
+
+	err = orm.RegisterDataBase("default", "postgres", dbURL)
+	if err != nil {
+		log.Critical(fmt.Sprintf("Database Registration failed: %v", err))
+	}
+
+	err = orm.RunSyncdb("default", false, true)
```
Comment (junan) [inline conf/initializers/database.go]: I think we don't need it as we added manual migration

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -34,12 +34,35 @@ jobs:
             echo "HEROKU_APP_NAME=${{ secrets.HEROKU_APP_NAME }}-staging" >> $GITHUB_ENV
           fi
 
+      - name: Set production environment variables
+        if: ${{ env.BRANCH_TAG == 'latest' }}
+        env:
+          DATABASE_URL: ${{ secrets.DATABASE_URL_PRODUCTION }}
+        run: |
+          echo "DATABASE_URL=$DATABASE_URL" >> $GITHUB_ENV
```
Comment (hoangmirs) [inline .github/workflows/deploy.yml:42]: I think we can get `DATABASE_URL` from Heroku variables, no?

File: Dockerfile [hunk 4]
```diff
@@ -9,6 +9,29 @@ COPY ./assets/. ./assets/
...
+    GOARCH=amd64
+
+# Copy the code into the container
+COPY . .
+
+# Install command-line tool
+RUN go get github.com/beego/bee/v2
+
+# Migrate database
+RUN bee migrate -driver=postgres -conn=$DATABASE_URL
```
Comment (hoangmirs) [inline Dockerfile:32]: Ah, because you're running migration inside the docker build

**Review by hoangmirs — APPROVED** [review]

**Review by junan — APPROVED** [review]

**Pull request merged**